src/audio/pcm_converter/pcm_converter_hifi3.c

```diff
@@ -658,12 +658,67 @@ static void pcm_convert_s16_to_f_lin(const void *psrc, void *pdst,
 	}
 }
 
+#if !defined(XT_ROUND_S)
+/**
+ * shift d value left (for positive a) or right (for negative a)
+ * and take care about overflows
+ */
+static inline int32_t _pcm_shift(int32_t d, int32_t a)
+{
+	int64_t dd = d;
+
+	if (a > 32)
+		a = 32;
+	else if (a < -32)
+		a = -32;
+
+	dd = a >= 0 ? dd << a : dd >> -a;
+	if (dd > INT32_MAX)
+		dd = INT32_MAX;
+
+	return (int32_t)dd;
+}
+/**
+ * \brief convert float number to fixed point
+ *
+ * Do not relay on compiler built-in float<=>int conversion in generic
+ * implementation, because "floating types float, double, and long double whose
+ * radix is not specified by the C standard but is usually two"
+ * ~https://gcc.gnu.org/onlinedocs/gcc/Decimal-Float.html
+ *
+ * \param src integer number to convert, it is int32_t to omit software float
+ *            operations library inclusion by compiler, when in whole topology
+ *            only external component needs float input.
+ * \param pow additional exponent component,
+ *            number of fractional bits in fixed point value.
+ *            Use '0' for normal conversion to integers
+ * \return (int32_t)src * 2**pow
+ */
+static int32_t _pcm_convert_f_to_i(int32_t src, int32_t pow)
+{
+	int32_t exponent, mantissa, dst;
```

Comment:
Code style, minor, we usually declare one variable at time.

Reply:
Will correct